### PR TITLE
refactor: Extract registry types in to own crate

### LINKED ISSUE #421: Create Control service
This service will be solely responsible for ensure the rest of the system is working as expected. It will read the current configuration from the Registry contract, and issue relevant commands to bring this configuration to life, i.e. it will Coordinate the rest of the system.

To allow an seamless transition from Coordinator V1 to V2, this should exist as a new service. We can call it `queryapi-coordinator-mainnet` to avoid conflicts.

Coordinator V2 will run an infinite loop consisting of the following logic:
  1. Fetch current indexers from registry contract
  2. Fetch Executor statuses
  3. Fetch Streamer statuses
  4. For each indexer
      - Provisioning
          - if the indexer has not been provisioned: provision it (we should only check once)
      - Streams
          - if there is no stream: start the stream from `last_processed_block`
          - If there is no stream and no `last_processed_block`, start from created/updated block
          - if stream exists and version differs: restart the stream from `start_block_height`
          - else: do nothing
      - Executors
          - if there is no executor: start it with current config
          - if executor exists and version differs: restart with new config
          - else: do nothing

### PR DESCRIPTION
**Pull request by morgsmccauley**
This PR extract the types exposed by the registry contract in to its own reusable crate: `registry_types`. This was done to avoid the duplication across our different service which; fetch the JSON, and then construct the Rust types.

## `registry_types` crate 
The registry types were already partially shared between Coordinator and the registry, but I've expanded on this to include all types rather than just a subset. I've created a new crate, rather than using the existing `./indexer/indexer_rule_type` crate as some packages needed to be upgraded, which caused conflicts across the `indexer/` cargo workspace - which seemed unnecessary to fix given that it is being deprecated.

## Registry contract updates
`near-sdk` version `4.1.1` is no longer compilable because it depends on a non-existent crate, hence the upgrade to `5.0.0-alpha.1`. Since I was already making changes here, I decided to remove the migration related code as it is no longer needed.

## Consuming `registry_types`
With the shared types in place, I've updated `block-streamer/` to use them, removing the existing types which it duplicates. I also went ahead and removed all types which were unused (left over from Alertexer).

`indexer_rule_type` depends on `near-sdk` version `4.1.1` (the broken one), and since it's no longer shared, I removed that dependency so that it can compile again.

Relates to https://github.com/near/queryapi/issues/421

### CODE REVIEW

File: registry/contract/Cargo.toml [hunk 1]
```diff
@@ -8,9 +8,10 @@ edition = "2021"
 crate-type = ["cdylib"]
 
 [dependencies]
-near-sdk = "4.1.1"
+borsh = "1.0.0"
+near-sdk = "5.0.0-alpha.1"
 uint = { version = "0.9.3", default-features = false }
-indexer_rule_type = { features = ["nearsdk"], path = "../../indexer/indexer_rule_type" }
+registry-types = { path = "../types", features = ["nearsdk"] }
```
Comment (roshaans) [inline registry/contract/Cargo.toml:14]: are we using the `nearsdk` feature?

**Comment by roshaans** [comment]
Can we add some details in the main readme about this `readme`?

**Comment by morgsmccauley** [comment]
> Can we add some details in the main readme about this `readme`?

The types are used internally, I'm not sure adding something would provide much benefit? Did you have something specific in mind?

File: registry/contract/Cargo.toml (hunk 1 above)
Comment (morgsmccauley) [reply registry/contract/Cargo.toml:14]: Yes - it forces the crate to use `near-sdk` types so that it is compatible with the registry contract

**Review by darunrs — APPROVED** [review]

**Review by roshaans — APPROVED** [review]

**Pull request merged**